parsl/monitoring/radios/zmq_router.py

```diff
@@ -129,3 +140,53 @@ def zmq_router_starter(*,
     else:
         comm_q.put(router.zmq_receiver_port)
         router.start()
+
+
+class ZMQRadioReceiver():
+    def __init__(self, *, process: SpawnProcessType, exit_event: EventType, port: int) -> None:
+        self.process = process
+        self.exit_event = exit_event
+        self.port = port
+
+    def close(self) -> None:
+        self.exit_event.set()
+        join_terminate_close_proc(self.process)
+
+
+def start_zmq_receiver(*,
+                       monitoring_messages: QueueType,
+                       loopback_address: str,
+                       port_range: Tuple[int, int],
+                       logdir: str,
+                       worker_debug: bool) -> ZMQRadioReceiver:
+    comm_q = SizedQueue(maxsize=10)
+
+    router_exit_event = SpawnEvent()
+
+    router_proc = SpawnProcess(target=zmq_router_starter,
+                               kwargs={"comm_q": comm_q,
+                                       "resource_msgs": monitoring_messages,
+                                       "exit_event": router_exit_event,
+                                       "address": loopback_address,
+                                       "port_range": port_range,
+                                       "run_dir": logdir,
+                                       "logging_level": logging.DEBUG if worker_debug else logging.INFO,
+                                       },
+                               name="Monitoring-ZMQ-Router-Process",
+                               daemon=True,
+                               )
+    router_proc.start()
+
+    try:
+        comm_q_result = comm_q.get(block=True, timeout=120)
+        comm_q.close()
+        comm_q.join_thread()
```

Comment:
Ought these be in the `finally` block?  Or otherwise no skipped in the exceptional case?